tests/test_util.go

```diff
@@ -231,9 +236,30 @@ func waitUntilVersionUp(cli driver.Client, allowNoLeaderResponse ...bool) error
 		return maskAny(noLeaderErr)
 	}
 
+	if predicate != nil {
```

Comment:
Move the predicate call inside the `op` function.
I expect the function to wait until the version response is valid, including any predicate I may pass in.